### ISSUE TITLE
Notifications do not show up in private channels

### ISSUE THREAD

**Issue by ryan-allen** [integrations/slack#677]
Heyo!

If we invite the github user to a private slack channel, and subscribe to a repo, notifications are not posted.

We are subscribing to an organisation repo, and the bot reports this:

`Subscribed #unknown-channel to org-name/repo-name`

If we subscribe in a public channel it works fine!

Also, hi! 😘

**Comment by stephen-bunn** [comment]
+1

**Comment by wilhelmklopp** [comment]
👋 

Just to confirm, it's not just the "Subscribed #unknown-channel to org-name/repo-name" message that shows up incorrectly but activity messages are actually not being delivered?

**Comment by wilhelmklopp** [comment]
Also could you confirm whether this was a public channel that was then converted to be a private channel?

**Comment by stephen-bunn** [comment]
Yes. None of the subscribed repositories have any notifications being delivered to the private channel.

The channel **was** a public channel that was converted to a private channel. However, the GitHub addon was not added until the repository was converted to be private.

**Comment by wilhelmklopp** [comment]
@stephen-bunn Thank you! ✨ 

I assume when you type `/github subscribe list` that shows all the subscriptions you would expect? Have you tried un and re-subscribing any of them?

When you type `/github debug` does the `"channel_id"` start with a "C" or with a "G"?

**Comment by stephen-bunn** [comment]
I've tried unsubbing and resubbing all of the repos we have connected and haven't seen any changes (no notifications being delivered). `/github subscribe list` does show me the repositories I have subscribed. But it reports `#unknown-channel is subscribed to` rather than the appropriate channel name.

The `channel_id` starts with a "C"

**Comment by wilhelmklopp** [comment]
Updating this with my findings:

I believe the underlying culprit is this change: https://twitter.com/SlackAPI/status/1037813133979443200

When I turn a public channel into a private channel, and then try to subscribe to a repo I can reproduce the `#unknown-channel is subscribed to` bug.

What I cannot reproduce is the activity messages not being delivered. See screenshot below:
![image](https://user-images.githubusercontent.com/7718702/45302421-31d35000-b50b-11e8-8b82-91fea2d5055c.png)

@stephen-bunn would you mind reaching out to [support](https://github.com/contact?form%5Bsubject%5D=GitHub%2BSlack%20integration) using the output from `/github debug`? We'll be able to dig into it more that way.

**Comment by stephen-bunn** [comment]
@wilhelmklopp Yeah sure no problem. I've just sent out the support form with the requested debug output.

**Comment by wilhelmklopp** [comment]
@stephen-bunn thank you!


@ryan-allen are you still seeing this problem or have activity messages started to show up for you?

**Comment by ryan-allen** [comment]
@wilhelmklopp whoops! sorry dropped the ball! i tested both with a _new private channel_ and a _public converted to private_, the converted one we tried to add the integration after it was private

i haven't tested it again, but i will! ❤

**Comment by DannyFeliz** [comment]
When I send a notification with the incoming webhook and the message contain a # with a word, it shows me #unknown-channel

**Comment by adimukherjee** [comment]
I'm seeing a similar issue with notifications for PRs showing up in public channels that have been subscribed to a repository and my DM with Github, but not in a private channel.
Rich previews are also only showing up on links posted by users who have signed in and enabled rich previews.

Was there any updated on this?

**Comment by fr-xushen** [comment]
I've seen a similar issue too.
The repository update will show in public channels or direct message, but not in a private channel.
It shows the configuration is done properly.

Appreciate any update on this.

**Comment by fr-xushen** [comment]
I found the solution.
If the channel is created as "private", github notification will show.
if the channle is changed from "public" to "private", it won't work.

**Comment by smarthive-skp** [comment]
I have a private channel and it is not showing notification. I get the "Either the app isn't installed on your repository or the repository does not exist. Install it to proceed" error.

**Comment by CyrilMazur** [comment]
> I found the solution.
> If the channel is created as "private", github notification will show.
> if the channle is changed from "public" to "private", it won't work.

You saved my day. I had exactly the same problem; Github notifications wouldn't show in a channel that used to be public and then converted to private.

I created a new private channel, added Github, subscribed to my repo, and then it worked.

**Comment by fr-xushen** [comment]
> > I found the solution.
> > If the channel is created as "private", github notification will show.
> > if the channle is changed from "public" to "private", it won't work.
> 
> You saved my day. I had exactly the same problem; Github notifications wouldn't show in a channel that used to be public and then converted to private.
> 
> I created a new private channel, added Github, subscribed to my repo, and then it worked.

Great, cheers~

**Comment by otto-mora** [comment]
So is the Github team thinking about solving this problem soon? I have the exact same problem where a channel was created as public and then changed to private. When connecting the Github App to Slack I get the following error after clicking "Connect Github account.

![image](https://user-images.githubusercontent.com/16960403/75174156-ba40b300-56f5-11ea-815b-1843c6f983af.png)

`An unexpected error occured. ref: 6d5667322f734793871c2c4cf44e53ee`

**Comment by guylepage3** [comment]
>I found the solution.
>If the channel is created as "private", github notification will show.
>if the channle is changed from "public" to "private", it won't work.

It would be nice fo users to switch back and forth between public and private. This was very confusing.

**Comment by fr-xushen** [comment]
Yes, cannot agree more.

**Comment by jvschoen** [comment]
Any update on this? we have history in these channels and would like to maintain the channels that were moved to private

**Comment by onsails** [comment]
Same problem happens after repository is renamed. Unsubscribe and subscribe doesn't fix the problem.

**Comment by onsails** [comment]
The problem appears when pushing to a non-master branch

**Comment by AbdelkhalekESI** [comment]
same problem here, no delivered notifications from GitHub when switching channel from public to private.

**Comment by ajbarry** [comment]
In our experience the case is that we have a channel that has always been private and we were getting repo activity notices as expected and enjoyed. The notices appears to have stopped "around the same time" as we invited an outside organization to join the channel. The organization never joined and the invitation has since expired.

All slash commands suggest the integration is configured correctly. I get a valid response to the `/github subscribe org/repo` command and if I immediately issue a `/github subscribe list` it does indicate we are subscribed to the repo in question.

However, coming back at some point later, like when you expect a github message, the `/github subscribe list` responds "Not subscribed to any repositories or accounts". I suspect a post attempt is made and fails, invalidating the subscription along with it.

**Comment by ajbarry** [comment]
Update: It was a bit heavy handed but I uninstalled the github app and added it back to the channel in question and github notices have resumed.

**Comment by matifreyre** [comment]
Same here, no notifications on a channel created public and switched to private. I've tried removing the app and adding it back, and also unsubscribing the repo and subscribing again.

**Comment by chinnichaitanya** [comment]
**Update as of 18th August 2020**

Even I have a situation where the notification stopped after converting a public channel to private. It worked after removing the Github app from the channel and reconfiguring everything again. 

It is a lot of work and it is easy to miss the details of our configurations if there are multiple applications. Definitely looking forward to these fixes since reinstalling and reconfiguring might not be feasible for everyone.

**Comment by nsangwan** [comment]
Same as a comment above, I've tried removing the app and adding it back, and also unsubscribing the repo and subscribing again. But I don't receive notifications.
Anyone on this thread who can give some definitive info on a solution?

**Comment by mwrona92** [comment]
We have a similar problem, it was working fine for years, then out of sudden it stopped. Unsubscribing and subscribing doesn't help.

**Comment by m8ow** [comment]
I faced same error.
I uninstalled Github App in Slack, and re-installing wouldn't work for me

**Comment by BaileySimrell** [comment]
So what's the solution for this? Has anybody had any luck?

**Comment by zwitterion** [comment]
After converting the public channel to private, notifications stopped working. Inviting @github solved the problem.

I entered these commands as a Slack message in the private channel:
/invite @github

not sure if these two are necessary but I also tried:
/github unsubscribe <your github org>
/github unsubscribe <your github org>

And after that, everything works. Looks like you no longer have to recreate the channel or change anything else.

**Comment by tcelestino** [comment]
@zwitterion, thank you so much for suggesting that, it really worked for me!
